blenderproc/python/loader/HavenMaterialLoader.py

```diff
@@ -114,26 +121,38 @@ def load_haven_mat(folder_path: str = "resources/haven", used_assets: Optional[L
     # makes the integration of complex materials easier
     addon_utils.enable("node_wrangler")
 
-    textures_folder_path = resolve_path(folder_path)
+    haven_folder = Path(resolve_path(str(folder_path)))
 
     if preload and fill_used_empty_materials:
         raise RuntimeError("Preload and fill used empty materials can not be done at the same time, check config!")
-        
-    if not os.path.exists(textures_folder_path) or not os.path.isdir(textures_folder_path):
-        raise FileNotFoundError("The folder path does not exist: {}".format(textures_folder_path))
 
-    texture_names = os.listdir(folder_path)
+    if not haven_folder.exists():
+        raise FileNotFoundError(f"The given haven folder does not exist: {haven_folder}")
+
+    # add the "textures" folder, if that is not already the case
+    if haven_folder.name != "textures" and (haven_folder / "textures").exists():
+        haven_folder = haven_folder / "textures"
+
+    texture_names: List[str] = os.listdir(haven_folder)
+    texture_names.sort()
+    if not texture_names:
+        raise FileNotFoundError(f"No texture folders found in {haven_folder}.")
+
+    # if only one element is returned
+    if return_random_element:
```

Comment:
The return_random_element option will probably not work together with used_assets, as the filtering happens after the random selection. So, if not by chance a texture from the used_assets is sampled, then this will throw an exception in line 186.
So either we document that or we move the used_assets filtering before the random seleciton.

Reply:
Agree, fixed.